core/src/main/scala/org/apache/spark/sql/crossdata/XDContext.scala

```diff
@@ -262,39 +272,94 @@ class XDContext protected (@transient val sc: SparkContext,
     *
     * @param path The local path or hdfs path where SparkContext will take the JAR
     */
-  override def addJar(path: String) = {
-    // TODO Add to current Classpath
+  def addJar(path: String, toClasspath:Option[Boolean]=None) = {
     super.addJar(path)
+    if (path.toLowerCase.startsWith("hdfs://")){
+      val hdfsIS: InputStream = HdfsUtils(xdConfig.getConfig(CoreConfig.HdfsKey)).getFile(path)
+      val file: java.io.File = createFile(hdfsIS, s"${xdConfig.getConfig(CoreConfig.JarsRepo).getString("externalJars")}/${path.split("/").last}")
+      if (toClasspath.getOrElse(true))
+        addToClasspath(file)
+    }else if (scala.reflect.io.File(path).exists){
+      val file=new java.io.File(path)
+      addToClasspath(file)
+    }else{
+      sys.error("File doesn't exist or is not a hdfs file")
+    }
+
+  }
+
+  private def addToClasspath(file:java.io.File): Unit = {
+    if (file.exists) {
+      val method: Method = classOf[URLClassLoader].getDeclaredMethod("addURL", classOf[URL])
+      method.setAccessible(true)
+      method.invoke(ClassLoader.getSystemClassLoader, file.toURI.toURL)
+      method.setAccessible(false)
+    } else {
+      sys.error(s"The file ${file.getName} not exists.")
+    }
+  }
+
+  private def createFile(hdfsIS: InputStream, path: String): java.io.File = {
+    val targetFile = new java.io.File(path)
+
+    val arrayBuffer = new Array[Byte](hdfsIS.available)
```

Comment:
This will read the file from HDFS and copy it entirely into memory. I'd rather write as the file is read. Something like:
`Files.copy(InputStream in, Path target)`
http://stackoverflow.com/a/19194580/1893995